src/Validator/SecurityTxtValidator.php

```diff
@@ -45,4 +50,24 @@ public function validate(SecurityTxt $securityTxt): SecurityTxtValidateResult
 		return new SecurityTxtValidateResult($errors, $warnings);
 	}
 
+
+	public function validateWithFetchResult(SecurityTxt $securityTxt, SecurityTxtFetchResult $fetchResult): SecurityTxtValidateResult
+	{
+		$result = $this->validate($securityTxt);
+		try {
+			$this->canonicalUrlValidator->validate($securityTxt, $fetchResult);
+		} catch (SecurityTxtError $e) {
+			return new SecurityTxtValidateResult(
+				array_merge($result->getErrors(), [$e->getViolation()]),
+				$result->getWarnings(),
+			);
+		} catch (SecurityTxtWarning $e) {
+			return new SecurityTxtValidateResult(
+				$result->getErrors(),
+				array_merge($result->getWarnings(), [$e->getViolation()]),
+			);
+		}
```

Comment:
This doesn't feel right, shouldn't be here, but you will already remove `validateWithFetchResult` anyway.